### DISCUSSION TITLE
Mapper is not running anymore

### DISCUSSION THREAD

**Discussion by joe-icanio** [srevinsaju/guiscrcpy#247] [Q&A]
Hi I am trying to use guiscrcpy but the mapper only worked for the first time. Can you help me I am new to this

**Comment by srevinsaju** [comment]
Hey, which OS are you using? Windows / Linux / MacOS / or anything else?

**Reply by joe-icanio** [reply]
Ubuntu 21.04

**Reply by joe-robin** [reply]
Is there anymore information I can give please help me

**Reply by joe-robin** [reply]
Please make sure that you have an X server running, and that the DISPLAY environment variable is set correctly
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 4369, resource id: 54525963, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 4378, resource id: 54525963, major code: 2 (ChangeWindowAttributes), minor code: 0

This is the error I get when clicking the mapper

I am using the guiscrcpy-.dev.r.glibc2.27-x86_64.AppImage

**Reply by srevinsaju** [reply]
can you send me the screenshot of your neofetch?

**Reply by joe-robin** [reply]
What is neofetch ? I dont know .

**Reply by joe-robin** [reply]
When I click initialise mapper button it doesnt do anything

**Reply by srevinsaju** [reply]
neofetch: https://github.com/dylanaraps/neofetch/wiki/Installation


how come you see the UI if you faced the previous error?

**Reply by joe-robin** [reply]
I see the UI but the mapper button only not working

**Reply by srevinsaju** [reply]
what is the error you see, in the Terminal?

**Reply by joe-robin** [reply]
```
qt.qpa.xcb: QXcbConnection: XCB error: 14 (BadIDChoice), sequence: 355, resource id: 1, major code: 53 (CreatePixmap), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 9 (BadDrawable), sequence: 356, resource id: 1, major code: 55 (CreateGC), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 13 (BadGC), sequence: 358, resource id: 41943049, major code: 60 (FreeGC), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 9 (BadDrawable), sequence: 359, resource id: 1, major code: 139 (Unknown), minor code: 4
qt.qpa.xcb: QXcbConnection: XCB error: 4 (BadPixmap), sequence: 360, resource id: 1, major code: 54 (FreePixmap), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 143 (Unknown), sequence: 361, resource id: 41943050, major code: 139 (Unknown), minor code: 27
qt.qpa.xcb: QXcbConnection: XCB error: 143 (Unknown), sequence: 362, resource id: 41943050, major code: 139 (Unknown), minor code: 7
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 365, resource id: 41943051, major code: 138 (Unknown), minor code: 23
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 366, resource id: 41943051, major code: 138 (Unknown), minor code: 23
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 367, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 368, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 414, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 415, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 563, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 576, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 592, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 605, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 643, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 654, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
scrcpy launched in 0.22s
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 790, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 791, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 838, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 839, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
/tmp/.mount_guiscrdUAX6I/scrcpy/usr/bin/scrcpy: /tmp/.mount_guiscrdUAX6I/scrcpy/usr/lib/libselinux.so.1: no version information available (required by /lib/x86_64-linux-gnu/libgio-2.0.so.0)
INFO: scrcpy 1.17 <https://github.com/Genymobile/scrcpy>
/tmp/.mount_guiscrdUAX6I/scrcpy/usr/lo...ped. 15.0 MB/s (34930 bytes in 0.002s)
[server] INFO: Device: vivo vivo 1909 (Android 11)
INFO: Renderer: opengl
INFO: OpenGL version: 3.0 Mesa 21.0.1
INFO: Trilinear filtering enabled
INFO: Initial texture: 1080x2400
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 994, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 1011, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 1035, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 1056, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
Traceback (most recent call last):
  File "guiscrcpy/launcher.py", line 271, in bootstrap_mapper
  File "/tmp/embedded.m5m_04kk.zip/shibokensupport/__feature__.py", line 142, in _import
  File "/usr/local/lib/python3.8/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 493, in exec_module
  File "guiscrcpy/lib/mapper/mapper.py", line 29, in <module>
  File "/tmp/embedded.m5m_04kk.zip/shibokensupport/__feature__.py", line 142, in _import
  File "/usr/local/lib/python3.8/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 493, in exec_module
  File "pynput/__init__.py", line 40, in <module>
  File "/tmp/embedded.m5m_04kk.zip/shibokensupport/__feature__.py", line 142, in _import
  File "/usr/local/lib/python3.8/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 493, in exec_module
  File "pynput/keyboard/__init__.py", line 31, in <module>
  File "pynput/_util/__init__.py", line 76, in backend
ImportError: this platform is not supported: No module named 'pynput.keyboard._xorg'

Try one of the following resolutions:

 * Please make sure that you have an X server running, and that the DISPLAY environment variable is set correctly
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 1107, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 1120, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
```

**Reply by joe-robin** [reply]
neofetch output
```
        `:+ssssssssssssssssss+:`           --------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 21.04 x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: H81M-S 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 5.11.0-25-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 5 hours, 16 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 2418 (dpkg), 12 (snap) 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 5.1.4 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1366x768 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: GNOME 3.38.4 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Mutter 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Dracula 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: gtk-master [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Dracula [GTK2/3] 
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: gnome-terminal 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: Intel i3-4130 (4) @ 3.400GHz 
    .ossssssssssssssssssdMMMNysssso.       GPU: Intel 4th Generation Core Proce 
      -+sssssssssssssssssyyyssss+-         Memory: 1733MiB / 3826MiB 
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.                                       
```

**Reply by srevinsaju** [reply]
Thanks for pasting the _entire_ error, now I can try to provide a solution:

What is the output of `echo $XDG_SESSION_TYPE`?

**Reply by joe-robin** [reply]
x11

Thank you for helping me

**Reply by srevinsaju** [reply]
@joe-robin Can you try out this version of the AppImage -> https://github.com/srevinsaju/guiscrcpy/suites/3323236556/artifacts/77858111

It will be a zip file, you will need to extract the appimage.

**Reply by joe-robin** [reply]
I tried it same results though

qt.qpa.xcb: QXcbConnection: XCB error: 14 (BadIDChoice), sequence: 355, resource id: 1, major code: 53 (CreatePixmap), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 9 (BadDrawable), sequence: 356, resource id: 1, major code: 55 (CreateGC), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 9 (BadDrawable), sequence: 357, resource id: 1, major code: 72 (PutImage), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 143 (Unknown), sequence: 362, resource id: 41943050, major code: 139 (Unknown), minor code: 7
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 365, resource id: 41943051, major code: 138 (Unknown), minor code: 23
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 366, resource id: 41943051, major code: 138 (Unknown), minor code: 23
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 367, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 368, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 414, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 415, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 485, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
scrcpy launched in 0.16s
/tmp/.mount_guiscrB9O1CU/scrcpy/usr/bin/scrcpy: /tmp/.mount_guiscrB9O1CU/scrcpy/usr/lib/libselinux.so.1: no version information available (required by /lib/x86_64-linux-gnu/libgio-2.0.so.0)
INFO: scrcpy 1.17 <https://github.com/Genymobile/scrcpy>
/tmp/.mount_guiscrB9O1CU/scrcpy/usr/lo...ped. 16.6 MB/s (34930 bytes in 0.002s)
[server] INFO: Device: vivo vivo 1909 (Android 11)
INFO: Renderer: opengl
INFO: OpenGL version: 3.0 Mesa 21.0.1
INFO: Trilinear filtering enabled
INFO: Initial texture: 2400x1080
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 703, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 712, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 754, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 755, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
Traceback (most recent call last):
  File "guiscrcpy/launcher.py", line 271, in bootstrap_mapper
  File "/tmp/embedded.vseyw8cw.zip/shibokensupport/__feature__.py", line 142, in _import
  File "/usr/local/lib/python3.8/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 493, in exec_module
  File "guiscrcpy/lib/mapper/mapper.py", line 29, in <module>
  File "/tmp/embedded.vseyw8cw.zip/shibokensupport/__feature__.py", line 142, in _import
  File "/usr/local/lib/python3.8/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 493, in exec_module
  File "pynput/__init__.py", line 41, in <module>
  File "/tmp/embedded.vseyw8cw.zip/shibokensupport/__feature__.py", line 142, in _import
  File "/usr/local/lib/python3.8/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 493, in exec_module
  File "pynput/mouse/__init__.py", line 29, in <module>
  File "pynput/_util/__init__.py", line 76, in backend
ImportError: this platform is not supported: No module named 'pynput.mouse._xorg'

Try one of the following resolutions:

 * Please make sure that you have an X server running, and that the DISPLAY environment variable is set correctly
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 810, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 819, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0

**Reply by srevinsaju** [reply]
Ok, I think the error has changed.

If you look closely, in your previous error, it was:
```
ImportError: this platform is not supported: No module named 'pynput.keyboard._xorg'
```
but now, it is
```
ImportError: this platform is not supported: No module named 'pynput.mouse._xorg'
```

So.. there is progress! 

![image](https://user-images.githubusercontent.com/48695438/126894455-8f99b47f-9cb5-4024-b9a6-94ca2ff52984.png)


Let me try to add a fix for mouse too, thanks for reporting this!!

**Reply by joe-robin** [reply]
LAMO

**Reply by srevinsaju** [reply]
This AppImage zip might work for you now:

https://github.com/srevinsaju/guiscrcpy/suites/3323461763/artifacts/77863096

**Reply by joe-robin** [reply]
by @srevinsaju
https://github.com/srevinsaju/guiscrcpy


qt.qpa.xcb: QXcbConnection: XCB error: 14 (BadIDChoice), sequence: 355, resource id: 1, major code: 53 (CreatePixmap), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 9 (BadDrawable), sequence: 356, resource id: 1, major code: 55 (CreateGC), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 13 (BadGC), sequence: 358, resource id: 41943049, major code: 60 (FreeGC), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 9 (BadDrawable), sequence: 359, resource id: 1, major code: 139 (Unknown), minor code: 4
qt.qpa.xcb: QXcbConnection: XCB error: 4 (BadPixmap), sequence: 360, resource id: 1, major code: 54 (FreePixmap), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 365, resource id: 41943051, major code: 138 (Unknown), minor code: 23
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 366, resource id: 41943051, major code: 138 (Unknown), minor code: 23
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 367, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 368, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 417, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 418, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
scrcpy launched in 0.16s
/tmp/.mount_guiscr0CQsIb/scrcpy/usr/bin/scrcpy: /tmp/.mount_guiscr0CQsIb/scrcpy/usr/lib/libselinux.so.1: no version information available (required by /lib/x86_64-linux-gnu/libgio-2.0.so.0)
INFO: scrcpy 1.17 <https://github.com/Genymobile/scrcpy>
/tmp/.mount_guiscr0CQsIb/scrcpy/usr/local/share/scrcpy/scrcpy-server: 1 file pushed, 0 skipped. 16.2 MB/s (34930 bytes in 0.002s)
[server] INFO: Device: vivo vivo 1909 (Android 11)
INFO: Renderer: opengl
INFO: OpenGL version: 3.0 Mesa 21.0.1
INFO: Trilinear filtering enabled
INFO: Initial texture: 2400x1080
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 654, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 674, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
[MAPPER] Waiting for device...
[MAPPER] Device connection established...
[MAPPER] Checking device orientation
Detected SurfaceOrientation, processing...
Detected Landscape orientation...
[SERVER] LISTENING VALUES:Your keys are being listened by server. 
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 739, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 740, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 795, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 845, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 855, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0
[MAPPER] Waiting for device...
[MAPPER] Device connection established...
[MAPPER] Checking device orientation
Detected SurfaceOrientation, processing...
Detected Landscape orientation...
[SERVER] LISTENING VALUES:Your keys are being listened by server. 
qt.qpa.xcb: QXcbConnection: XCB error: 6 (BadCursor), sequence: 900, resource id: 41943051, major code: 2 (ChangeWindowAttributes), minor code: 0


I still dont see the mapper

**Reply by srevinsaju** [reply]
Well, looks like the mapper works now.. according to the logs. The mapper is not visible normally, it just listens to the keyboard shortcuts which you are typing realtime, and performs a click on the android screen. I think you missed the mapper interactive configuration part. To see it again, you will have to

1. Close the current running guiscrcpy 
2. Open terminal and type 
```bash
rm -r ~/.config/guiscrcpy/guiscrcpy.mapper.json
```
3. Start guiscrcpy appimage again.

It should work then.

**Reply by joe-robin** [reply]
Traceback (most recent call last):
  File "guiscrcpy/launcher.py", line 317, in bootstrap_mapper
  File "guiscrcpy/lib/utils.py", line 180, in open_process
  File "subprocess.py", line 858, in __init__
  File "subprocess.py", line 1704, in _execute_child
FileNotFoundError: [Errno 2] No such file or directory: '/home/joe/Downloads/Compressed/guiscrcpy-r-continuous-x86_64'

**Reply by joe-robin** [reply]
I ran that command now it displayed a message I once say when I first tried it but now it says like this.

**Reply by srevinsaju** [reply]
Did you run the AppImage after extracting it?

**Reply by joe-robin** [reply]
yes

**Reply by srevinsaju** [reply]
Are you running the appimage from the terminal? or double-clicked from the file manager?

**Reply by joe-robin** [reply]
I put the app image in a different folder and ran it. Now it works I am running it on the terminal

Thanks for the help

**Reply by srevinsaju** [reply]
Sure, anytime :D

**Reply by joe-robin** [reply]
It says the mapping are successful but the keys are not woring when I tried it. Am I missing something ?
